### ISSUE TITLE
internal error with self relation query

### ISSUE THREAD

**Issue by steebchen** [prisma/prisma#3354]
## Bug description

self relation query doesn't work

## How to reproduce

```graphql
query {
  findManyPost(where: { morePosts: { some: { title: { equals: "my" } } } }) {
    id
    title
    userId
    postId
  }
}
```

results in 

```
{
  "errors": [
    {
      "error": "Error occurred during query execution:\nConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: Db, cause: Some(DbError { severity: \"ERROR\", parsed_severity: Some(Error), code: SqlState(\"42601\"), message: \"syntax error at or near \\\")\\\"\", detail: None, hint: None, position: Some(Original(259)), where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some(\"scan.l\"), line: Some(1149), routine: Some(\"scanner_yyerror\") }) }) })",
      "user_facing_error": {
        "is_panic": false,
        "message": "Error occurred during query execution:\nConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: Db, cause: Some(DbError { severity: \"ERROR\", parsed_severity: Some(Error), code: SqlState(\"42601\"), message: \"syntax error at or near \\\")\\\"\", detail: None, hint: None, position: Some(Original(259)), where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some(\"scan.l\"), line: Some(1149), routine: Some(\"scanner_yyerror\") }) }) })",
        "backtrace": null
      }
    }
  ]
}
```

## Prisma information

```prisma
generator photon {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DB")
}

model User {
  id    String @default(cuid()) @id
  posts Post[]
}

model Post {
  id        String  @default(cuid()) @id
  title     String
  morePosts Post[]  @relation("PostToPost")

  User      User?   @relation(fields: [userId], references: [id])
  userId    String?
  Post      Post?   @relation("PostToPost", fields: [postId], references: [id])
  postId    String?
}
```

## Environment & setup

- OS: 
- Database:  postgres
- Node.js version: 
- Prisma version:

```
@prisma/cli          : 2.5.0
Current platform     : darwin
Query Engine         : query-engine 9a670138b1db276001d785a2adcba1584c869d24 (at node_modules/@prisma/cli/query-engine-darwin)
Migration Engine     : migration-engine-cli 9a670138b1db276001d785a2adcba1584c869d24 (at node_modules/@prisma/cli/migration-engine-darwin)
Introspection Engine : introspection-core 9a670138b1db276001d785a2adcba1584c869d24 (at node_modules/@prisma/cli/introspection-engine-darwin)
Format Binary        : prisma-fmt 9a670138b1db276001d785a2adcba1584c869d24 (at node_modules/@prisma/cli/prisma-fmt-darwin)
Studio               : 0.259.0
```

originally reported at https://github.com/prisma/prisma-client-go/issues/206

**Comment by pantharshit00** [comment]
I can reproduce this using the javascript client. Here is a minimal reproduction: https://github.com/harshit-test-org/prisma-issue-3354/


![image](https://user-images.githubusercontent.com/22195362/91041543-df26cc00-e62d-11ea-8cf6-27ad36d4482c.png)

**Comment by pantharshit00** [comment]
This has been fixed in version `2.7.0`
